[BUG] Drop bcrypt for bcryptjs

### Describe the bug
Bcrypt causes errors when self-signed certificates in use. This is not apparent until self-signed certificates or proxy in use.

- Drop bcrypt
- Add bcryptjs

> bcryptjs is same as bcrypt, but optimized with zero dependencies.

### To Reproduce
Steps to reproduce the behavior:
1. Add a self-signed certificate to Windows 10
2. Fork and clone repo
3. npm install
4. npm run test
5. notice error in image

### Expected behavior
Tests run with no errors

### Screenshots
![image](https://user-images.githubusercontent.com/17856665/86761180-b034a680-c03d-11ea-8988-c7477b91b3de.png)


### Desktop (please complete the following information):
 - OS: [Windows 10]
 - Node v12.6.2

### Smartphone (please complete the following information):

### Additional context

```bash
$ npm install bcrypt
npm WARN registry Unexpected warning for https://registry.npmjs.org/: Miscellaneous Warning UNABLE_TO_ET_ISSUER_CERT_LOCALLY: request to https://registry.npmjs.org/bcrypt failed, reason: unable to get locl issuer certificate
npm WARN registry Using stale data from https://registry.npmjs.org/ due to a request error during revaidation.

> bcrypt@5.0.0 install C:\Users\piouson\source\repos\hng\comment-microapi\node_modules\bcrypt
> node-pre-gyp install --fallback-to-build

node-pre-gyp WARN Using request for node-pre-gyp https download 
node-pre-gyp WARN Pre-built binaries not installable for bcrypt@5.0.0 and node@12.16.2 (node-v72 ABI, nknown) (falling back to source compile with node-gyp) 
node-pre-gyp WARN Hit error unable to get local issuer certificate
gyp WARN install got an error, rolling back install
gyp ERR! configure error 
gyp ERR! stack Error: unable to get local issuer certificate
gyp ERR! stack     at TLSSocket.onConnectSecure (_tls_wrap.js:1474:34)
gyp ERR! stack     at TLSSocket.emit (events.js:310:20)
gyp ERR! stack     at TLSSocket._finishInit (_tls_wrap.js:917:8)
gyp ERR! stack     at TLSWrap.ssl.onhandshakedone (_tls_wrap.js:687:12)
gyp ERR! System Windows_NT 10.0.18363
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\piouson\\AppData\\Roaming\\npm\\nodemodules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "configure" "--fallback-to-build" "--module=C:\Users\\piouson\\source\\repos\\hng\\comment-microapi\\node_modules\\bcrypt\\lib\\binding\\napi-v3\\bcrpt_lib.node" "--module_name=bcrypt_lib" "--module_path=C:\\Users\\piouson\\source\\repos\\hng\\comment-icroapi\\node_modules\\bcrypt\\lib\\binding\\napi-v3" "--napi_version=5" "--node_abi_napi=napi" "--nap_build_version=3" "--node_napi_label=napi-v3"
gyp ERR! cwd C:\Users\piouson\source\repos\hng\comment-microapi\node_modules\bcrypt
gyp ERR! node -v v12.16.2
gyp ERR! node-gyp -v v5.0.7
gyp ERR! not ok
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute 'C:\Program Files\nodejs\node.exe C:\Users\piouson\Appata\Roaming\npm\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js configure --fallback-to-build -module=C:\Users\piouson\source\repos\hng\comment-microapi\node_modules\bcrypt\lib\binding\napi-v3\bcryp_lib.node --module_name=bcrypt_lib --module_path=C:\Users\piouson\source\repos\hng\comment-microapi\nod_modules\bcrypt\lib\binding\napi-v3 --napi_version=5 --node_abi_napi=napi --napi_build_version=3 --nod_napi_label=napi-v3' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (C:\Users\piouson\source\repos\hng\comment-micrapi\node_modules\node-pre-gyp\lib\util\compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:310:20)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:1021:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)  
node-pre-gyp ERR! System Windows_NT 10.0.18363
node-pre-gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\piouson\\source\\repos\\hng\comment-microapi\\node_modules\\node-pre-gyp\\bin\\node-pre-gyp" "install" "--fallback-to-build"     
node-pre-gyp ERR! cwd C:\Users\piouson\source\repos\hng\comment-microapi\node_modules\bcrypt
node-pre-gyp ERR! node -v v12.16.2
node-pre-gyp ERR! node-pre-gyp -v v0.15.0
node-pre-gyp ERR! not ok
Failed to execute 'C:\Program Files\nodejs\node.exe C:\Users\piouson\AppData\Roaming\npm\node_modules\nm\node_modules\node-gyp\bin\node-gyp.js configure --fallback-to-build --module=C:\Users\piouson\source\epos\hng\comment-microapi\node_modules\bcrypt\lib\binding\napi-v3\bcrypt_lib.node --module_name=bcryptlib --module_path=C:\Users\piouson\source\repos\hng\comment-microapi\node_modules\bcrypt\lib\binding\nai-v3 --napi_version=5 --node_abi_napi=napi --napi_build_version=3 --node_napi_label=napi-v3' (1)      
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! bcrypt@5.0.0 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the bcrypt@5.0.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.    

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\piouson\AppData\Roaming\npm-cache\_logs\2020-07-07T09_08_48_225Z-debug.log
```

--- a/package.json
+++ b/package.json
@@ -33,7 +33,7 @@
   "homepage": "https://github.com/microapidev/comment-microapi#readme",
   "dependencies": {
     "@hapi/joi": "^17.1.1",
-    "bcrypt": "^5.0.0",
+    "bcryptjs": "^2.4.3",
     "cookie-parser": "^1.4.5",
     "cors": "^2.8.5",
     "dotenv": "^8.2.0",

--- a/utils/auth/passwordUtils.js
+++ b/utils/auth/passwordUtils.js
@@ -1,4 +1,4 @@
-const bcrypt = require("bcrypt");
+const bcrypt = require("bcryptjs");
 
 exports.hashPassword = async (password) => {
   const hash = await bcrypt.hash(password, 10);

